src/RedisClusterResourceManager.php

```diff
@@ -54,29 +54,25 @@ private function createRedisResource(RedisClusterOptions $options): RedisCluster
                 $options->getTimeout(),
                 $options->getReadTimeout(),
                 $options->isPersistent(),
-                $options->getPassword(),
+                $options->getAuthentication(),
```

Comment:
Please assemble the `authentication` argument here rather than outsourcing this to the `RedisClusterOptions`.
Maybe at some point, the extension will change to different formats - I rather have all the version specific logic in the resource manager rather than within options. That is an implementation detail which should not be part of a class called "options".